Lib/aifc.py

```diff
@@ -179,23 +181,26 @@ def _read_string(file):
         dummy = file.read(1)
     return data
 
-_HUGE_VAL = 1.79769313486231e+308 # See <limits.h>
-
 def _read_float(f): # 10 bytes
-    expon = _read_short(f) # 2 bytes
+    data = f.read(10)
+    if len(data) != 10:
+        raise EOFError
+    expon = int.from_bytes(data[:2], 'big', signed=True)
     sign = 1
     if expon < 0:
         sign = -1
         expon = expon + 0x8000
-    himant = _read_ulong(f) # 4 bytes
-    lomant = _read_ulong(f) # 4 bytes
-    if expon == himant == lomant == 0:
+    mant = int.from_bytes(data[2:], 'big')
+    if not expon and not mant:
```

Comment:
Is this special case necessary? I think the right thing to do here is to say: `if expon == 0: expon = 1`, and then continue to the `ldexp` call; I _think_ this will do the right thing both on current Python, and on some future hypothetical version of Python where `float` does actually have the range required to represent the subnormal range of the x87 extended precision format.

Reply:
Sorry, but I don't understand the reason of changing 0 to 1 in exponent.